compiler/resolution/nilChecking.cpp

```diff
@@ -782,6 +993,23 @@ static void gatherVariablesToCheck(FnSymbol* fn,
   }
 }
 
+// The formals are initially unknown.
+// When we are confindent in our nilable type-checking, set the formals
+// whose types are non-nilable to MUST_ALIAS_ALLOCATED.
```

Comment:
I think this should set formals of non-nilable class type to MUST_ALIAS_ALLOCATED now - see my earlier comment on the PR

Reply:
I added a discussion of this to the PR message and kept "unknown" for now.